### PR TITLE
Add armv8m architecture support

### PR DESCRIPTION
**Pull request by agkaminski**
JIRA: RTOS-881

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (mcxn94x).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
https://github.com/phoenix-rtos/phoenix-rtos-project/pull/1120
https://github.com/phoenix-rtos/plo/pull/337
https://github.com/phoenix-rtos/phoenix-rtos-utils/pull/222
https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/569
https://github.com/phoenix-rtos/phoenix-rtos-hostutils/pull/55
https://github.com/phoenix-rtos/phoenix-rtos-filesystems/pull/132
https://github.com/phoenix-rtos/phoenix-rtos-devices/pull/508
https://github.com/phoenix-rtos/phoenix-rtos-build/pull/195
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: arch/arm/v8m/reboot.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+{
+	platformctl_t pctl = {
+		.action = pctl_get,
+		.type = pctl_reboot,
+	};
+
+	*val = 0;
+	if (platformctl(&pctl) < 0)
+		return -1;
+
```
Comment (lukileczo) [inline arch/arm/v8m/reboot.c:52]:
```suggestion
	if (platformctl(&pctl) < 0) {
		return -1;
	}
```

Comment (agkaminski) [reply arch/arm/v8m/reboot.c]: Fixed, thx

**Review by lukileczo — APPROVED** [review]
LGTM

**Review by lukileczo — APPROVED** [review]
LGTM

**Pull request merged**